src/librustc_mir/interpret/machine.rs

```diff
@@ -184,6 +185,22 @@ pub trait Machine<'mir, 'tcx>: Sized {
         dest: PlaceTy<'tcx, Self::PointerTag>,
     ) -> InterpResult<'tcx>;
 
+    /// Called to read the specified `local` from the `frame`.
+    fn access_local(
+        _ecx: &InterpCx<'mir, 'tcx, Self>,
+        frame: &Frame<'mir, 'tcx, Self::PointerTag, Self::FrameExtra>,
+        local: mir::Local,
+    ) -> InterpResult<'tcx, Operand<Self::PointerTag>> {
+        frame.locals[local].access()
+    }
+
+    /// Called before a `StaticKind::Static` value is accessed.
```

Comment:
Please specify what "access" means here. Is this called when the address of a static is taken? Or just on reads/writes?

Reply:
I will double check and improve the comment in the other open PR I have but I believe this is just on reads/writes.